Heffner_Daniel/Rails Section 1/product_dashboard_2/controllers/products_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ProductsController < ApplicationController
+  def index
+    @products = Product.all
+  end
+
+  def show
+    @product = Product.find(params[:id])
+    @comment = Comment.new
+  end
+
+  def new
+    @product = Product.new
+    @categories = Category.all
+  end
+
+  def edit
+    @product = Product.find(params[:id])
+    @errors = Product.find(params[:id])
+    @categories = Category.all
+  end
+
+  def create
+    p_params = product_params
```

Comment:
I'd call it something like attributes, mostly that is a personal preference. Something about calling them params when they aren't anymore and the prefix.